docs_source/content/resource-authors-guide.md

```diff
@@ -0,0 +1,131 @@
+pattern is most obvious in the shell task resource:
+
+```hcl
+task "example" {
+  check = "test -f hello.txt"
+  apply = "touch hello.txt"
+}
+```
+
+When Converge applies this resource, the check statement is run first. If the
+exit code is non-zero, Converge then runs the apply statement. Afterwards, it
+runs the check statement again to see if the application was successful.
+
+To implement something like the shell task, you would start by implementing a
+struct and it's corresponding methods.
+[`resource.Task`](https://godoc.org/github.com/asteris-llc/converge/resource#Task)
+is the interface you'll need to implement.
+
+```go
+struct MyShellTask struct {
+    resource.Status
+
+    CheckStmt string
+    ApplyStmt string
+}
+
+func (t *MyShellTask) Check(resource.Renderer) (resource.TaskStatus, error) {
+    // your check implementation, returning t (which embeds a TaskStatus)
+}
+
+func (t *MyShellTask) Apply() (resource.TaskStatus, error) {
+    // your apply implementation, returning t (which embeds a TaskStatus)
+}
+```
+
+### Task Status
+
+Check and Apply both return
+[`TaskStatus`](https://godoc.org/github.com/asteris-llc/converge/resource#TaskStatus).
+You can implement that interface yourself, but the most common pattern is to
+embed
+[`Status`](https://godoc.org/github.com/asteris-llc/converge/resource#Status) in
+the task itself. This is good for two reasons: first, this makes the fields on
+your struct available for lookups. Second, and more importantly, `Status`
+provides a number of helper methods to make your integration go more smoothly.
+
+`Status` has three fields: `Differences`, `Output`, and `Level`. They all have
+accurate documentation on their fields, which we will not repeat here. However,
+the interaction between `Differences` and `Level` deserves special mention.
+These two fields are how you control execution of your Apply method. They follow
+these rules:
+
+1. If the level is equal to
+   [`resource.StatusWillChange`](https://godoc.org/github.com/asteris-llc/converge/resource#StatusLevel)
+   or
+   [`resource.StatusCantChange`](https://godoc.org/github.com/asteris-llc/converge/resource#StatusLevel),
+   the Status will always show up as having changes.
+1. Otherwise, if there are any diffs which say that they contain a difference,
+   the Status will always show as having changes.
+
+## Preparer
+
+Before you can use your resource, it has to be deserialized from HCL. For this,
+we will write a
+[`resource.Resource`](https://godoc.org/github.com/asteris-llc/converge/resource#Resource).
+Resource exists to render a Task's fields and return it in it's executable
+state. In our case, our Resource would look like this (they're typically called
+`Preparer`).
+
+```go
+type Preparer struct {
+    Check string `hcl:"check"`
+    Apply string `hcl:"apply"`
+}
+
+func (p *Preparer) Render(r *resource.Renderer) (resource.Task, error) {
+    check, err := render.Render("check", p.Check)
```

Comment:
We should maybe mention what `render.Render` is doing here, and explain the expectation that render errors are returned.  It's maybe an API smell but right now we're reliant on the fact that a preparer will return `render.ErrUnresolvable` if that's what's returned from `render.Render` in order to understand when we need to defer node evaluation.